src/aap_eda/services/ruleset/activation_podman.py

```diff
@@ -195,6 +149,98 @@ def run_worker_mode(
+        heartbeat: int,
+        ports: dict,
+    ) -> Container:
+        """Run ansible-rulebook in worker mode."""
+        args = [
+            "ansible-rulebook",
+            "--worker",
+            "--websocket-ssl-verify",
+            ws_ssl_verify,
+            "--websocket-address",
+            ws_url,
+            "--id",
+            str(activation_instance.id),
+            "--heartbeat",
+            str(heartbeat),
+        ]
+        if (
+            settings.ANSIBLE_RULEBOOK_LOG_LEVEL
+            and settings.ANSIBLE_RULEBOOK_LOG_LEVEL in VALID_LOG_LEVELS
+        ):
+            args.append(settings.ANSIBLE_RULEBOOK_LOG_LEVEL)
+
+        self.pod_args["name"] = f"eda-{activation_instance.id}-{uuid.uuid4()}"
+        if ports:
+            self.pod_args["ports"] = ports
+        self._load_extra_args()
+        self.activation_db_logger.write("Starting Container", True)
+        self.activation_db_logger.write(f"Container args {args}", True)
+        container = self.client.containers.run(
+            image=self.decision_environment.image_url,
+            command=args,
+            stdout=True,
+            stderr=True,
+            remove=True,
+            detach=True,
+            **self.pod_args,
+        )
+
+        logger.info(
+            f"Created container: "
+            f"name: {container.name}, "
+            f"id: {container.id}, "
+            f"ports: {container.ports}, "
+            f"status: {container.status}, "
+            f"command: {args}"
+        )
+
+        now = timezone.now()
+        activation_instance.status = ActivationStatus.RUNNING
+        activation_instance.updated_at = now
+        activation_instance.activation_pod_id = container.id
+        activation_instance.save(
+            update_fields=["status", "updated_at", "activation_pod_id"]
+        )
+
+        activation = activation_instance.activation
+        activation.status = ActivationStatus.RUNNING
+        activation.is_valid = True
+        activation.status_updated_at = now
+        activation.save(
+            update_fields=[
+                "status",
+                "status_updated_at",
+                "is_valid",
+                "modified_at",
+            ]
+        )
+        return container
+
+    def _fetch_container(
+        self, activation_instance: models.ActivationInstance
+    ) -> Container:
+        containers = self.client.containers.list(
+            all=True, filters={"id": activation_instance.activation_pod_id}
+        )
+        if not containers:
+            raise ReconnectContainerNotFound(
+                f"Container {activation_instance.activation_pod_id} not found"
+            )
+        container = containers[0]
+        if container.attrs["State"] != "running":
+            raise ReconnectContainerNotFound(
+                f"Container {activation_instance.activation_pod_id} not in "
+                "running state"
+            )
+        return container
```

Comment:
IMO the responsibility of this method is confusing. It is raising an exception if the container is not in running state. I feel at least should be named "_fetch_running_container" or it should be in charge of just fetching the container, the logic to raise ReconnectContainerNotFound should be executed by the caller when it is called under the block "if reconnect:" which makes more obvious that it is a logic related with the reconnection.